kolibri/core/device/api.py

```diff
@@ -164,3 +172,48 @@ def patch(self, request):
         settings.name = request.data["name"]
         settings.save()
         return Response({"name": settings.name})
+
+
+class SyncStatusFilter(FilterSet):
+
+    member_of = ModelChoiceFilter(
+        method="filter_member_of", queryset=Collection.objects.all()
+    )
+
+    def filter_member_of(self, queryset, name, value):
+        return queryset.filter(
+            Q(user__memberships__collection=value) | Q(user__facility=value)
+        )
+
+    class Meta:
+        model = UserSyncStatus
+        fields = ["user", "member_of"]
+
+
+class UserSyncStatusViewSet(ReadOnlyValuesViewset):
+    permission_classes = (KolibriAuthPermissions,)
+    filter_backends = (KolibriAuthPermissionsFilter, DjangoFilterBackend)
+    queryset = UserSyncStatus.objects.all()
+    filter_class = SyncStatusFilter
+
+    values = (
+        "id",
+        "queued",
+        "sync_session",
+        "last_synced",
+        "active",
+        "user",
+        "user_id",
+    )
+
+    def get_queryset(self):
+        return UserSyncStatus.objects.filter()
+
+    def annotate_queryset(self, queryset):
+
+        queryset = queryset.annotate(
```

Comment:
These feel right, we can double check after merging to see if doing this or an explicit subquery produces a more performant query.